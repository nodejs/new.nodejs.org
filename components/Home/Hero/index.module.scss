@use '../../../styles/typography.scss';

.hero {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: var(--space-96) 0 var(--space-128) 0;
  text-align: center;

  h1,
  p {
    margin: 0;
  }

  h1 {
    max-width: 840px;

    @media (max-width: 900px) {
      font-size: var(--font-size-display2);
      line-height: 48px;
      max-width: 90vw;
    }
  }

  .subTitle {
    @extend .t-subheading;

    color: var(--color-text-secondary);
    font-weight: var(--font-weight-light);
    margin: var(--space-32) 0 var(--space-32) 0;
    max-width: 780px;

    @media (max-width: 900px) {
      max-width: 90vw;
    }
  }

  .buttonsContainer {
    align-items: flex-start;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;

    .downloadLtsContainer {
      align-items: center;
      display: flex;
      flex-direction: column;
      position: relative;

      & > p {
        margin-top: var(--space-08);
      }

      @media (max-width: 640px) {
        margin-bottom: var(--space-20);
      }
    }
  }
}

.downloadButton {
  align-items: center;
  background-color: var(--brand6);
  border-radius: 5rem;
  box-sizing: border-box;
  color: var(--black0);
  display: flex;
  flex-direction: column;
  font-weight: var(--font-weight-semibold);
  height: 5.6rem;
  justify-content: center;
  margin: 0 var(--space-16);
  padding: 2.4rem;
  text-decoration: none;
  transition: background-color 0.2s ease-out;

  span {
    color: var(--black4);
    font-size: var(--font-size-body3);
  }

  &:hover {
    background-color: var(--brand7);
    color: var(--black0);
  }

  &:focus {
    @extend .downloadButton;
  }

  &Inverse {
    @extend .downloadButton;

    background-color: transparent;
    border: var(--brand6) var(--space-02) solid;
    color: var(--brand6);
    transition: background-color 0.2s ease-out;

    &:hover {
      background-color: var(--brand7);
      border: var(--brand7) var(--space-02) solid;
    }
  }
}

[data-theme='dark'] .downloadButtonInverse {
  color: var(--black0);
}
