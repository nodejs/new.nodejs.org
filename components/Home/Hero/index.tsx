import Link from 'next/link';
import { FormattedMessage } from 'react-intl';
import styles from './index.module.scss';
import { useDetectOS } from '../../../hooks/useDetectOS';
import type { FC } from 'react';
import type { NodeReleaseData } from '../../../types';

type HeroProps = {
  title: string;
  subTitle?: string;
  nodeReleaseData: NodeReleaseData[];
};

const Hero: FC<HeroProps> = ({ title, subTitle, nodeReleaseData }) => {
  const { getDownloadLink } = useDetectOS();

  const currentLTS = nodeReleaseData.find(release => release.isLts);
  const currentRelease = nodeReleaseData.find(
    release => release.status === 'Current'
  );

  const ltsVersionUrl = getDownloadLink(currentLTS?.fullVersion || '');
  const currentVersionUrl = getDownloadLink(currentRelease?.fullVersion || '');
  return (
    <div className={styles.hero}>
      <h1>{title}</h1>
      <h2 className={styles.subTitle}>{subTitle}</h2>
      <div className={styles.buttonsContainer}>
        <div className={styles.downloadLtsContainer}>
          <a className={styles.downloadButton} href={ltsVersionUrl}>
            <FormattedMessage id="components.hero.downloadLts" />
            <span>
              <FormattedMessage
                id="components.hero.currentVersion"
                values={{ version: currentLTS?.fullVersion }}
              />
            </span>
          </a>
          <p className="t-caption">
            <a href={currentVersionUrl}>
              <FormattedMessage
                id="components.hero.getCurrent"
                values={{ version: currentRelease?.fullVersion }}
              />
            </a>
          </p>
        </div>
        <Link href="/learn" className={styles.downloadButtonInverse}>
          <FormattedMessage id="components.hero.learn" />
        </Link>
      </div>
    </div>
  );
};

export default Hero;
