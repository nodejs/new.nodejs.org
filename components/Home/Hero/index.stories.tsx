import Hero from '.';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof Hero>;
type Meta = MetaObj<typeof Hero>;

export const Default: Story = {
  args: {
    title: 'Run JavaScript Everywhere.',
    subTitle:
      'Node.js is a free, open-sourced, cross-platform JavaScript run-time environment that lets developers write command line tools and server-side scripts outside of a browser.',
    nodeReleaseData: [
      {
        version: '18',
        fullVersion: 'v18.15.0',
        codename: '',
        isLts: true,
        status: 'Active LTS',
        initialRelease: '',
        ltsStart: '',
        maintenanceStart: null,
        endOfLife: '',
      },
      {
        version: '19',
        fullVersion: 'v19.8.1',
        codename: '',
        isLts: true,
        status: 'Current',
        initialRelease: '',
        ltsStart: '',
        maintenanceStart: null,
        endOfLife: '',
      },
    ],
  },
};

export default { component: Hero } as Meta;
