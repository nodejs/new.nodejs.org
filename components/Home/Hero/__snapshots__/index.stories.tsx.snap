// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/Hero Default smoke-test 1`] = `
<div class="Hero_hero__M5nJ2">
  <h1>
    Run JavaScript Everywhere.
  </h1>
  <h2 class="Hero_subTitle__fW9Mo">
    Node.js is a free, open-sourced, cross-platform JavaScript run-time environment that lets developers write command line tools and server-side scripts outside of a browser.
  </h2>
  <div class="Hero_buttonsContainer__f5dSG">
    <div class="Hero_downloadLtsContainer__46E7X">
      <a class="Hero_downloadButton__QeN_O"
         href="https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg"
      >
        Download Node (LTS)
        <span>
          Version v18.15.0
        </span>
      </a>
      <p class="t-caption">
        <a href="https://nodejs.org/dist/v19.8.1/node-v19.8.1.pkg">
          Download Latest (v19.8.1)
        </a>
      </p>
    </div>
    <a class="Hero_downloadButtonInverse__Dbhon"
       href="/learn"
    >
      Learn Node
    </a>
  </div>
</div>
`;
